src/network/core/http_winhttp.cpp

```diff
@@ -171,20 +173,22 @@ void NetworkHTTPRequest::WinHttpCallback(DWORD code, void *info, DWORD length)
 		case WINHTTP_CALLBACK_STATUS_SECURE_FAILURE:
 		case WINHTTP_CALLBACK_STATUS_REQUEST_ERROR:
 			Debug(net, 0, "HTTP request failed: {}", GetLastErrorAsString());
-			this->callback->OnFailure();
 			this->finished = true;
+			this->callback->OnFailure();
 			break;
 
 		default:
 			Debug(net, 0, "HTTP request failed: unexepected callback code 0x{:x}", code);
-			this->callback->OnFailure();
 			this->finished = true;
+			this->callback->OnFailure();
 			return;
 	}
 }
 
 static void CALLBACK StaticWinHttpCallback(HINTERNET handle, DWORD_PTR context, DWORD code, void *info, DWORD length)
 {
+	if (context == NULL) return;
```

Comment:
Maybe `nullptr` makes mingw32 happy?

Reply:
a `nullptr` is not valid, as it is a `DWORD_PTR`. So .. I am afraid I will have to use `0` .........